src/Corax/Utils/GrowableHashSet.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+
+namespace Corax.Utils;
+
+public sealed class GrowableHashSet<TItem>
```

Comment:
why do we have this? So we can handle > 2B items to find?

If so, increasing when the count is at `int.MaxValue` is _super_ late, you already hit the max fill factor.

Reply:
Yes, that was the reason why we created it. Previously, it was in Raven.Server, but I moved that code to Corax.

Reply:
So it needs to be viable, what if we'll use something like this:
https://gist.github.com/ayende/c0586f194d37db5262f07aca5c7c6341

roughly based on roaring bitmaps

Reply:
Agree

Reply:
Having said that, handling that in a separate PR is best